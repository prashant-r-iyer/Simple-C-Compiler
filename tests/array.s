	.text

.globl main
main:
	# Save Frame pointer
	pushq %rbp
	movq %rsp, %rbp
	subq $256, %rsp
	pushq %rbx
	pushq %rbx
	pushq %r10
	pushq %r13
	pushq %r14
	pushq %r15
	pushq %r11
	pushq %r9

	# push 8
	movq $8, %rbx

	# push 20
	movq $20, %r10

	# *
	imulq %r10, %rbx
	# func=malloc nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	call malloc
	movq %rax, %rbx
	movq %rbx, -8(%rbp)

	# push 8
	movq $8, %rbx

	# push 9
	movq $9, %r10
	imulq $8, %rbx
	addq -8(%rbp), %rbx
	movq %r10, (%rbx)
	movq $string0, %rbx

	# push 8
	movq $8, %r10
	imulq $8, %r10
	addq -8(%rbp), %r10
	movq (%r10), %r10
	# func=printf nargs=2

	# Move values from reg stack to reg args
	movq %r10, %rsi
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

	# Restore registers
	popq %r9
	popq %r11
	popq %r15
	popq %r14
	popq %r13
	popq %r10
	popq %rbx
	popq %rbx
	addq $256, %rsp

	leave
	ret

string0:
	.string "a[8]=%d\n"

