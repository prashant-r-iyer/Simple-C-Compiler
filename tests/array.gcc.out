array.c: In function ‘main’:
array.c:6:13: warning: implicit declaration of function ‘malloc’ [-Wimplicit-function-declaration]
    6 |         a = malloc(8*20);
      |             ^~~~~~
array.c:1:1: note: include ‘<stdlib.h>’ or provide a declaration of ‘malloc’
  +++ |+#include <stdlib.h>
    1 | 
array.c:6:13: warning: incompatible implicit declaration of built-in function ‘malloc’ [-Wbuiltin-declaration-mismatch]
    6 |         a = malloc(8*20);
      |             ^~~~~~
array.c:6:13: note: include ‘<stdlib.h>’ or provide a declaration of ‘malloc’
array.c:9:9: warning: implicit declaration of function ‘printf’ [-Wimplicit-function-declaration]
    9 |         printf("a[8]=%d\n", a[8]);
      |         ^~~~~~
array.c:1:1: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
  +++ |+#include <stdio.h>
    1 | 
array.c:9:9: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
    9 |         printf("a[8]=%d\n", a[8]);
      |         ^~~~~~
array.c:9:9: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
array.c:9:23: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘long int’ [-Wformat=]
    9 |         printf("a[8]=%d\n", a[8]);
      |                      ~^     ~~~~
      |                       |      |
      |                       int    long int
      |                      %ld
